exporters/ostream/include/opentelemetry/exporters/ostream/metric_exporter.h

```diff
@@ -72,8 +73,8 @@ class OStreamMetricExporter final : public opentelemetry::sdk::metrics::PushMetr
 
 private:
   std::ostream &sout_;
-  bool is_shutdown_ = false;
-  mutable opentelemetry::common::SpinLockMutex lock_;
+  std::atomic<bool> is_shutdown_{false};
+  std::mutex serialize_lock_;
```

Comment:
Why not using SpinLockMutext here?

Reply:
We were unfortunately using the `lock_` not just for shutdown :)

Reply:
The SpinLockMutex was also used to serialize when writing to sout_.

When the disk is full, a write or flush can hold forever, and using spin locks can make threads pilling up in CPU loops, which is very bad.

A mutex is more appropriate.